src/libraries/Common/tests/System/Net/Http/Http2LoopbackConnection.cs

```diff
@@ -263,15 +306,25 @@ public void IgnoreWindowUpdates()
             _ignoreWindowUpdates = true;
         }
 
-        // Set up loopback server to expect PING frames among other frames.
+        // Set up loopback server to expect a (non-ACK) PING frame among other frames.
         // Once PING frame is read in ReadFrameAsync, the returned task is completed.
         // The returned task is canceled in ReadPingAsync if no PING frame has been read so far.
         public Task<PingFrame> ExpectPingFrameAsync()
         {
             _expectPingFrame ??= new TaskCompletionSource<PingFrame>();
+
             return _expectPingFrame.Task;
         }
 
+        // Recurring variant of ExpectPingFrame().
+        // Starting from the time of the call, respond to all (non-ACK) PING frames which are received among other frames.
+        private void SetuptransparentPingResponse()
+        {
+            if (_transparentPingResponse) return;
+            _transparentPingResponse = true;
+            _ = ExpectPingFrameAsync();
```

Comment:
Why call ExpectPingFrameAsync here? All it does is create a Task you then ignore.

Reply:
I think it would be cleaner to just explicitly check for _transparentPingResponse in ReadFrameAsync. So we don't need to deal with _expectPingFrame at all here.